File: Easy/lettercase_percentage_ratio.py
data = ['thisTHIS', 'AAbbCCDDEE', 'N', 'UkJ']

import string
#print string.ascii_lowercase
#print string.ascii_uppercase

#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	test = test.rstrip()
	whole = 0
	lower = 0
	upper = 0
	for l in test:
		whole += 1
		if l in string.ascii_lowercase:
			lower += 1
		elif l in string.ascii_uppercase:
			upper += 1
	L = (float(lower) / whole) * 100
	U = (float(upper) / whole) * 100
	print("lowercase: %.2f uppercase: %.2f" % (L, U))

#test_cases.close()

File: Easy/lettercase_percentage_ratio.r
test.cases <- c("thisTHIS", "AAbbCCDDEE",
				"N", "UkJ")

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		words <- strsplit(test, "")[[1]]
		whole = length(words)
		lower_part = length(words[words %in% letters])
		upper_part = length(words[words %in% LETTERS])
		L = format(round((lower_part / whole)*100, digits = 2), nsmall = 2)
		U = format(round((upper_part / whole)*100, digits = 2), nsmall = 2)
		paste(c("lowercase:", L, "uppercase:", U), sep = " ", collapse = " ")
	}
}), sep = "\n")